sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -103,10 +108,11 @@ export class Receiver {
    * @returns {ReceiveHandler} ReceiveHandler - An object that provides a mechanism to stop receiving more messages.
    */
   receive(onMessage: OnMessage, onError: OnError, cancellationToken?: Aborter): ReceiveHandler {
-    const sReceiver = StreamingReceiver.create(this._context, this.partitionId, this._receiverOptions);
-    sReceiver.prefetchCount = Constants.defaultPrefetchCount;
-    this._context.receivers[sReceiver.name] = sReceiver;
-    return sReceiver.receive(onMessage, onError);
+    this._throwIfReceiverOrConnectionClosed();
+    this._streamingReceiver = StreamingReceiver.create(this._context, this.partitionId, this._receiverOptions);
+    this._streamingReceiver.prefetchCount = Constants.defaultPrefetchCount;
+    this._context.receivers[this._streamingReceiver.name] = this._streamingReceiver;
```

Comment:
Looks like `StreamingReceiver.create()` function takes care of setting the receiver in the `_context.receivers` map. So we don't need this step